packages/main/src/ToolbarItem.ts

```diff
@@ -107,10 +132,27 @@ class ToolbarItem extends UI5Element {
 			},
 		};
 	}
+
+	constructor() {
+		super();
+	}
+
+	/**
+	 * Handles the click event on the toolbar item.
+	 * If `preventOverflowClosing` is false, it will fire a "close-overflow" event.
+	 */
+	onClick(e: Event): void {
+		e.stopImmediatePropagation();
```

Comment:
We stop the propagation here which may lead to apps not being able to handle the click event on a higher level container. If possible, let's try to avoid that.